### DISCUSSION TITLE
Move .flake8 config to pyproject.toml

### DISCUSSION THREAD

**Discussion by AlexanderHott** [hikari-py/hikari#1377] [Ideas]
As of flake8 0.5.0 [(hikari uses 0.5.4)](https://github.com/hikari-py/hikari/blob/master/dev-requirements/flake8.txt#L1), you are able to define the flake8 configuration inside of the pyproject.toml file via the [Flake8-pyproject](https://pypi.org/project/Flake8-pyproject/) flake8 plugin. 

This upgrade drops support for [flake8-broken-line](https://pypi.org/project/flake8-broken-line/), but black already removes any backslashes.

> Black prefers parentheses over backslashes, and will remove backslashes if found.
 [(source)](https://black.readthedocs.io/en/stable/the_black_code_style/current_style.html)

The .flake8 file [currently lists IFSTMT001 as one of the ignored codes](https://github.com/hikari-py/hikari/blob/master/.flake8#L13), which I think should be [IFS001](https://github.com/d1618033/flake8-if-statements#checks), but hikari doesn't even use that flake8 plugin so it can just be removed. The 6 letter then 3 number code causes Flake8-pyproject to error, that's why I'm bringing this up.

All other tool configs are defined in the pyproject.toml, so I see no issue in moving it. I'd be happy to create a PR if this idea gets approved.

**Comment by davfsa** [comment]
> The .flake8 file [currently lists IFSTMT001 as one of the ignored codes](https://github.com/hikari-py/hikari/blob/master/.flake8#L13), which I think should be [IFS001](https://github.com/d1618033/flake8-if-statements#checks), but hikari doesn't even use that flake8 plugin so it can just be removed. The 6 letter then 3 number code causes Flake8-pyproject to error, that's why I'm bringing this up.

This is just a left-over from when we had the plugin enabled

> The 6 letter then 3 number code causes Flake8-pyproject to error, that's why I'm bringing this up.

That is the reason we initially disabled it.

> All other tool configs are defined in the pyproject.toml, so I see no issue in moving it. I'd be happy to create a PR if this idea gets approved.

Feel free to!

**Comment by FasterSpeeding** [comment]
This should be marked as resolved or whatever since flake8 isn't being used anymore

**Comment by davfsa** [comment]
Flake8 has been switched to ruff, and ruff has its own config file too :)
